pkg/policy/selectorcache.go

```diff
@@ -105,15 +108,24 @@ type SelectorCache struct {
 	startNotificationsHandlerOnce sync.Once
 }
 
+// GetCurrentVersionHold returns a VersoionHold for the current version.
+// The returned VersionHold must be closed with Close()
```

Comment:
I wonder if we can add a lint rule that detects un-closed holds? I'm not familiar with this sort of analysis.

Reply:
I added Endpoint.buildLock/Unlock functions that are used instead of calling `e.buildMutex.Lock` or `e.buildMutex.Unlock` directly, and added "linting" there. IMO the build mutex unlock is the perfect spot for this.